<center><h1>Técnicas de Inteligencia Artificial</header1></center>

Presentado por: Genesis Belen Heredia  <br>
Fecha: 17/07/2022

# Importación de librerias necesarias

In [16]:
#Para esta actividad se importarán las siguientes librerías:
#Se implementan las librerias para obtener los datos

#Se realiza la importación de la libreria pandas
import pandas as pd

#Se realiza la importación de la libreria sklearn
from sklearn.model_selection import train_test_split
from sklearn import metrics

#Se realiza la importación de la libreria statsmodels
import statsmodels.api as smi

#Se realiza la importación de la libreria necesaria para la regresión lineal
from sklearn.linear_model import LinearRegression 

#Se realiza la importación de la libreria que permite realizar la grafica
import matplotlib.pyplot as plt
from sklearn.preprocessing import LabelEncoder

## Cargar el Dataset

In [17]:
#Código para cargar Dataset
#Se cargan los datos del Dataset mediante la libreria pandas
FishLinear = pd.read_csv("Fish.csv" )

In [20]:
#Se visualizan 10 registros
FishLinear.head(15)

Species  Weight  Length1  Length2  Length3   Height   Width
0    Bream   242.0     23.2     25.4     30.0  11.5200  4.0200
1    Bream   290.0     24.0     26.3     31.2  12.4800  4.3056
2    Bream   340.0     23.9     26.5     31.1  12.3778  4.6961
3    Bream   363.0     26.3     29.0     33.5  12.7300  4.4555
4    Bream   430.0     26.5     29.0     34.0  12.4440  5.1340
5    Bream   450.0     26.8     29.7     34.7  13.6024  4.9274
6    Bream   500.0     26.8     29.7     34.5  14.1795  5.2785
7    Bream   390.0     27.6     30.0     35.0  12.6700  4.6900
8    Bream   450.0     27.6     30.0     35.1  14.0049  4.8438
9    Bream   500.0     28.5     30.7     36.2  14.2266  4.9594
10   Bream   475.0     28.4     31.0     36.2  14.2628  5.1042
11   Bream   500.0     28.7     31.0     36.2  14.3714  4.8146
12   Bream   500.0     29.1     31.5     36.4  13.7592  4.3680
13   Bream   340.0     29.5     32.0     37.3  13.9129  5.0728
14   Bream   600.0     29.4     32.0     37.2  14.9544  5.1708

## Preprocesamiento del dataset. Transformaciones previas necesarias para la modelación


LabelEncoder

In [22]:
#Transformación de las caracteristicas (mire el apartado Feature engineering del aula virtual)
encoder = LabelEncoder()

In [24]:
FishLinear["Species"] = encoder.fit_transform(FishLinear["Species"])

In [25]:
FishLinear.head()

Species  Weight  Length1  Length2  Length3   Height   Width
0        0   242.0     23.2     25.4     30.0  11.5200  4.0200
1        0   290.0     24.0     26.3     31.2  12.4800  4.3056
2        0   340.0     23.9     26.5     31.1  12.3778  4.6961
3        0   363.0     26.3     29.0     33.5  12.7300  4.4555
4        0   430.0     26.5     29.0     34.0  12.4440  5.1340

In [26]:
#Función que permite describir el dataset
FishLinear.describe()

Species       Weight     Length1     Length2     Length3  \
count  159.000000   159.000000  159.000000  159.000000  159.000000   
mean     2.264151   398.326415   26.247170   28.415723   31.227044   
std      1.704249   357.978317    9.996441   10.716328   11.610246   
min      0.000000     0.000000    7.500000    8.400000    8.800000   
25%      1.000000   120.000000   19.050000   21.000000   23.150000   
50%      2.000000   273.000000   25.200000   27.300000   29.400000   
75%      3.500000   650.000000   32.700000   35.500000   39.650000   
max      6.000000  1650.000000   59.000000   63.400000   68.000000   

           Height       Width  
count  159.000000  159.000000  
mean     8.970994    4.417486  
std      4.286208    1.685804  
min      1.728400    1.047600  
25%      5.944800    3.385650  
50%      7.786000    4.248500  
75%     12.365900    5.584500  
max     18.957000    8.142000

## Descripción de la fuente del Dataset

In [27]:
#Código que responde a la descripción anterior
FishLinear.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 159 entries, 0 to 158
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Species  159 non-null    int32  
 1   Weight   159 non-null    float64
 2   Length1  159 non-null    float64
 3   Length2  159 non-null    float64
 4   Length3  159 non-null    float64
 5   Height   159 non-null    float64
 6   Width    159 non-null    float64
dtypes: float64(6), int32(1)
memory usage: 8.2 KB


***Descripción***

Mediante el uso de la función info podemos deducir que existen 159 registros los cuales representan a diferentes caracteristicas de. El dataset cuenta con 7 campos donde se puede tomar como variable objetivo **(Weight)** y como variables de entrada:

**Variables de entrada X** Variable Independiente

**-->Species:** Representa el numero de la casa(Variable Continua)

**-->Length1** Representa la fecha de compra (Variable continua)

**-->Length2** Representa la edad que tiene el establecimiento (Variable Continua)

**-->Length3** Representa la distancia a la estación MRT más cercana (Variable Continua)

**-->Height** número de tiendas de conveniencia (Variable Continua)

**-->Widht** Latitud (Variable continua)



**Variable de salida Y** Variable Dependiente

**--> Weight:** peso del pescado


## Explique el problema a resolver. 
Descripción del problema. Tipo de problema (justifique). Variable objetivo, variables de entrada. Utilidad de su posible solución. Elementos adicionales que considere relevantes (no son necesarios contenidos teóricos, sino explicar qué relaciones tratas de comprobar y con qué métodos).

## Caracterización del Dataset

Realice una descripción de los datos con:

>- Número de instancias en total.
>- Número de atributos de entrada, su significado y tipo.
>- Estadísticas de la variable objetivo.
>- Estadísticas los atributos en relación con la variable objetivo.


In [5]:
#Código que responde a la descripción anterior

En un par de párrafos haga un resumen de los principales hallazagos encontrados:    

## División del dataset en datos de entrenamiento y datos de test 

In [5]:
#Código que realice la división en entrenamiento y test, de acuerdo con la estretgia de evaluación planeada. Describa cuál es.


## Regresión Lineal

In [6]:
#Códo del modelo

In [7]:
#Entrenamiento

In [8]:
#Test

## Evaluación del Modelo

Construya un o dos párrafos con los principales hallazgos. Cómo está funcionando el modelo? Calidad en los resultados de predicción. 